### PR TITLE
release: 10.12.2

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
base branch `10.12`  was created to have a base for this PR .... no real need to merge this into `10.12`

but after release we need to merge this back into `master`- will do a new PR once we are there ....

**Review by phil-davis — COMMENTED** [review]
LGTM
should we bump `version.php` in this PR, or do it in a separate PR to the 10.12 branch?
(and @jnweiger will need to run the changelog... stuff anyway from the 10.12 branch when preparing the release)

File: apps/federatedfilesharing/lib/FedShareManager.php [hunk 1]
```diff
@@ -328,8 +320,9 @@ public function updateOcmPermissions(IShare $share, $ocmPermissions) {
 	 *
 	 * @return void
 	 */
-	public function updatePermissions(IShare $share, $permissions) {
-		if ($share->getPermissions() !== $permissions) {
+	public function updatePermissions(IShare $share, int $permissions): void {
+		# permissions can only be reduced but not upgraded
+		if (Permissions::isNewPermissionHigher($share->getPermissions(), $permissions)) {
```
Comment (phil-davis) [inline apps/federatedfilesharing/lib/FedShareManager.php:325]: See my comment on the master PR https://github.com/owncloud/core/pull/40803/files#r1212913427

File: apps/federatedfilesharing/lib/Ocm/Permissions.php [hunk 2]
```diff
@@ -77,4 +77,8 @@ public function toOcPermissions($ocmPermissions) {
 
 		return $ocPermissions;
 	}
+
+	public static function isNewPermissionHigher(int $existingPermission, int $newPermission): bool {
+		return ($existingPermission | $newPermission) === $existingPermission;
+	}
```
Comment (phil-davis) [inline apps/federatedfilesharing/lib/Ocm/Permissions.php:83]: See my comment on the master PR
https://github.com/owncloud/core/pull/40803/files#r1212917165

File: changelog/unreleased/40803 [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Bugfix: Disallow permission tobe upgraded via federated sharing
```
Comment (phil-davis) [inline changelog/unreleased/40803:1]:
```suggestion
Bugfix: Disallow permission to be upgraded via federated sharing
```

**Comment by phil-davis** [comment]
The code in this `release/10.12.2` branch has been released.
It has also been applied to `master` - see PR #40828 for the details.

CI fails because it messes around trying to auto-merge stuff
https://drone.owncloud.com/owncloud/core/38565
```
+ git checkout master
Branch 'master' set up to track remote branch 'master' from 'origin'.
Already on 'master'
+ git fetch origin refs/pull/40827/head:
From https://github.com/owncloud/core
 * branch                  refs/pull/40827/head -> FETCH_HEAD
+ git merge 01ece70d7519e6fbea5b11fd5183c84c795e6862
Auto-merging version.php
CONFLICT (content): Merge conflict in version.php
Auto-merging tests/acceptance/features/bootstrap/OccContext.php
Auto-merging tests/acceptance/features/bootstrap/EncryptionContext.php
Auto-merging sonar-project.properties
CONFLICT (content): Merge conflict in sonar-project.properties
Auto-merging core/Command/Encryption/EncryptAll.php
Auto-merging changelog/10.12.1_2023-04-03/40709
CONFLICT (rename/delete): changelog/unreleased/40702 deleted in 01ece70d7519e6fbea5b11fd5183c84c795e6862 and renamed to changelog/10.12.1_2023-04-03/40702 in HEAD. Version HEAD of changelog/10.12.1_2023-04-03/40702 left in tree.
Auto-merging changelog/10.12.1_2023-04-03/40697
Auto-merging changelog/10.12.1_2023-04-03/40693
Auto-merging apps/federatedfilesharing/tests/NotificationsTest.php
Auto-merging apps/federatedfilesharing/tests/Controller/RequestHandlerTest.php
Auto-merging CHANGELOG.md
CONFLICT (content): Merge conflict in CHANGELOG.md
Auto-merging .drone.star
Automatic merge failed; fix conflicts and then commit the result.
```

Maybe we should force-merge this PR - that will get the `10.12` branch head to point to the same commit as the `v10.12.2` tag, and that should make it easy and understandable for the next person, if there turns out to be any need for `10.12.3` !

**Review by phil-davis — APPROVED** [review]

**Comment by pako81** [comment]
I guess this is not needed anymore, isn't it?

**Comment by phil-davis** [comment]
> I guess this is not needed anymore, isn't it?

correct - not needed. The code in this branch was also added to master by PR #40828

**Pull request closed without merging**